lib/scss_lint/linter/merge_equal_selector.rb

```diff
@@ -0,0 +1,60 @@
+module SCSSLint
+  class Linter::MergeEqualSelector < Linter
+    include LinterRegistry
+
+    def check_node(node)
+      visits = Set.new
+      node.children.each do |child_node|
+
+        equal_node = find_equal_node child_node, node.children
+        next unless equal_node
+        next if visits.include? [child_node, equal_node] or
+                visits.include? [equal_node, child_node]
+        visits << [child_node, equal_node]
+
+        add_lint child_node.line,
+        "Merge rule `#{node_rule(child_node)}` on line #{child_node.line} with rule " \
+        "on line #{equal_node.line}" do
```

Comment:
This is incredibly hard to read. Breaking out the message into a separate variable so we can write:

```ruby
message = "..."
add_lint message do
  ...
end
```

...makes it much easier to read the block.